packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -293,27 +297,31 @@ class Auth {
      */
     async ready() {
         if (this.fetchedToken && this.fetchedToken !== '') {
+            console.log('--- fetched token:', this.fetchedToken)
             const {isGuest, customerId, usid} = this.parseSlasJWT(this.fetchedToken)
             this.set('access_token', this.fetchedToken)
             this.set('customer_id', customerId)
             this.set('usid', usid)
             this.set('customer_type', isGuest ? 'guest' : 'registered')
-            this.pendingToken = Promise.resolve(this.data)
-            return this.pendingToken
+            return Promise.resolve(this.data)
         }
+        // TODO: in my scenario, the code exits early here. It should continue on and refresh the token.
         if (this.pendingToken) {
+            console.log('--- return pending token')
             return this.pendingToken
         }
         const accessToken = this.get('access_token')
+        console.log('--- got access token:', accessToken)
 
         if (accessToken && !this.isTokenExpired(accessToken)) {
-            this.pendingToken = Promise.resolve(this.data)
-            return this.pendingToken
+            console.log('--- has access token that is not expired')
+            return Promise.resolve(this.data)
```

Comment:
No longer setting `pendingToken` here because it's meant to be set when making an API request. Not when the token is already available.